README.md

```diff
@@ -191,6 +192,30 @@ import localeData from "moment/locale/zh-cn.js";
 import fp from "lodash/fp.js";
 ```
 
+#### General URL-like specifier remapping
+
+As part of allowing general remapping of specifiers, import maps specifically allow remapping of URL-like specifiers, such as `"https://example.com/foo.mjs"` or `"./bar.mjs"`. One of the more advanced usages of this is [for fallbacks](#for-built-in-modules-in-browsers-without-import-maps), but here we demonstrate some basic ones to communicate the concept:
+
+```json
+{
+  "imports": {
+    "https://www.unpkg.com/vue/dist/vue.runtime.esm.js": "/node_modules/vue/dist/vue.runtime.esm.js"
+  }
+}
+```
+
+This remapping ensures that any imports of the unpkg.com version of Vue (at least at that URL) instead grab the one from the local server.
+
+```json
+{
+  "imports": {
+    "/app/helpers.mjs": "/app/helpers/index.mjs"
+  }
+}
+```
+
+This remapping ensures that any URL-like imports that resolve to `/app/helpers.mjs`, including e.g. an `import "./helpers.mjs"` from files inside `/app/`, or an `import "../helpers.mjs"` from files inside `/app/models`, will instead resolve to `/app/helpers/index.mjs`. This is probably not a good idea; instead of creating an indirection which obfuscates your code, you should instead just update your source files to import the correct files. But, it is a useful example for demonstrating the capabilities of import maps.
```

Comment:
> This remapping ensures that any URL-like imports that resolve to `/app/helpers.mjs`, including e.g. an `import "./helpers.mjs"` from files inside `/app/`, or an `import "../helpers.mjs"` from files inside `/app/models`, will instead resolve to `/app/helpers/index.mjs`.

Kind of repeating a comment in another thread - this suggests a "partial resolution" step that seems to be missing from the spec so far. It sounds like the idea is:

1. If `url` starts with `./` or `../`, set `url` to `new URL(url, referrer).path`.
2. Apply import mappings to `url`.
3. Assert: `url` isn't a bare specifier.
3. Return `new URL(url, referrer).href`.

Is that about correct?

Reply:
More or less. See https://github.com/WICG/import-maps/tree/master/reference-implementation

Reply:
Ah, I missed the fact that the import map gets preprocessed in `parser.js` where `/foo/bar` keys get turned into fully resolved URL keys (if I'm scanning it correctly). Thanks for pointing me in the right direction. :)